Rakefile

```diff
@@ -1,9 +1,28 @@
 require 'rake'
 require 'rake/testtask'
 
-desc 'rake with no argument will run "rake test"'
-task default: :test
+require_relative 'lib/tasks/exercise'
+require_relative 'lib/tasks/exercise_test_tasks'
+require_relative 'lib/tasks/rake_arguments'
```

Comment:
> Should exercise_test_tasks require them, even though it doesn't use any of those classes?

Well, There's a philosophical starting point for you, the question becomes why not?

1. When RakeArguments goes away and options becomes a variable that problem goes away.

2. You need `Exercise` so you can call `Exercise.all` as an argument to `ExericiseTestTasks.new` and when defining the `exericses` task, but I would argue that defining the task to run all the exercises belongs in `ExerciseTestTasks` also.

And maybe `exercises:` should not be an argument to `ExerciseTestTasks.new` at all.

When you take care of these 2 things, the `Exercise` dependency disappears and everything becomes more elegant, which suggests to me it's the right thing to do.